docker/daemon/client.go

```diff
@@ -0,0 +1,65 @@
+package daemon
+
+import (
+	"github.com/containers/image/types"
+	"github.com/docker/docker/api"
+	dockerclient "github.com/docker/docker/client"
+	"github.com/docker/go-connections/tlsconfig"
+	"net/http"
+	"path/filepath"
+)
+
+// NewDockerClient initializes a new API client based on the passed SystemContext.
+func newDockerClient(systemCtx *types.SystemContext) (*dockerclient.Client, error) {
+	if systemCtx == nil {
+		return dockerclient.NewClient(dockerclient.DefaultDockerHost, api.DefaultVersion, nil, nil)
+	}
+
+	httpClient, err := tlsConfig(systemCtx)
+	if err != nil {
+		return nil, err
+	}
+
+	host := systemCtx.DockerHost
+	if host == "" {
+		host = dockerclient.DefaultDockerHost
+	}
+
+	version := systemCtx.DockerAPIVersion
+	if version == "" {
+		version = api.DefaultVersion
+	}
+
+	cli, err := dockerclient.NewClient(host, version, httpClient, nil)
+	if err != nil {
+		return cli, err
+	}
+
+	return cli, nil
+}
+
+func tlsConfig(systemCtx *types.SystemContext) (*http.Client, error) {
+	if systemCtx == nil || systemCtx.DockerCertPath == "" {
+		return nil, nil
+	}
+
+	options := tlsconfig.Options{
+		CAFile:             filepath.Join(systemCtx.DockerCertPath, "ca.pem"),
+		CertFile:           filepath.Join(systemCtx.DockerCertPath, "cert.pem"),
+		KeyFile:            filepath.Join(systemCtx.DockerCertPath, "key.pem"),
+		InsecureSkipVerify: systemCtx.DockerTLSVerify,
```

Comment:
**NO** this inverts the semantics.

Also, the default (i.e. when the boolean is not set by the user in `types.SystemContext`, must be be secure; i.e. the `SystemContext` member should be `…SkipTLSVerify`, preferably with the `Insecure` warning in there as well.

Reply:
ok

Reply:
The confusion came from me trying to mimic/align with DOCKER_TLS_VERIFY. Logic is inverted here. I'll add the 'Skip'